libafl_frida/src/helper.rs

```diff
@@ -1400,24 +1399,38 @@ impl<'a> FridaInstrumentationHelper<'a> {
     > {
         // We only care for compare instrunctions - aka instructions which set the flags
         match instr.mnemonic().unwrap() {
-            "cmp" | "ands" | "subs" | "adds" | "negs" | "ngcs" | "sbcs" | "bics" | "cls"
-            | "cbz" | "tbz" | "tbnz" => (),
+            "cmp" | "ands" | "subs" | "adds" | "negs" | "ngcs" | "sbcs" | "bics" | "cbz"
+            | "cbnz" | "tbz" | "tbnz" | "adcs" => (),
             _ => return Err(()),
         }
-        let operands = self
+        let mut operands = self
             .capstone
             .insn_detail(instr)
             .unwrap()
             .arch_detail()
             .operands();
 
         // cbz - 1 operand, tbz - 3 operands
-        let special_case = ["cbz", "tbz", "tbnz"].contains(&instr.mnemonic().unwrap());
-        if operands.len() != 2 || !special_case {
+        let special_case = [
+            "cbz", "cbnz", "tbz", "tbnz", "subs", "adds", "ands", "sbcs", "bics", "adcs",
+        ]
+        .contains(&instr.mnemonic().unwrap());
+        if operands.len() != 2 && !special_case {
             return Err(());
         }
+
+        // handle special opcodes case which have 3 operands, but the 1st(dest) is not important to us
+        if vec!["subs", "adds", "ands", "sbcs", "bics", "adcs"].contains(&instr.mnemonic().unwrap())
```

Comment:
I think this can just be a slice instead of a `vec`, so just use `["subs", "adds", "ands", "sbcs", "bics", "adcs"]` directly.
Building up a `vec` may lead to an allocation each time (that may be optimized away, but who knows)

Reply:
ok :)